packages/workflow-diagram/src/index.tsx

```diff
@@ -28,7 +29,16 @@ const WorkflowDiagram: React.FC<{
 
   useEffect(() => {
     if (projectSpace) {
-      Store.setProjectSpace(projectSpace);
+      const jobs = addDescription(projectSpace.jobs);
+      const workflows = renameUntitledWorkflows(projectSpace.workflows);
+
+      const transformedProjectSpace = {
+        ...projectSpace,
+        jobs,
+        workflows,
+      } as ProjectSpace;
+
+      Store.setProjectSpace(transformedProjectSpace);
```

Comment:
Instead of changing the ProjectSpace, I think the description should be computed in the Node.

Have a look at `layout/factories.ts`, you can attach things to the `__flowProps__` objects and handle them in the components. This way we aren't breaking the contract of the ProjectSpace type by manipulating it (adding description) and we open up the pathway for avoiding re-renders on the individual nodes (_we don't currently do anything about this now, but when/if we do this pattern helps_).

Reply:
Done.